PyInstaller/utils/hooks/__init__.py

```diff
@@ -496,12 +490,26 @@ def django_dottedstring_imports(django_root_dir):
     # Many times Django users do not specify absolute imports in the settings module.
     pths.append(django_root_dir)
 
-    package_name = os.path.basename(django_root_dir) + '.settings'
-    env = {'DJANGO_SETTINGS_MODULE': package_name,
-           'PYTHONPATH': os.pathsep.join(pths)}
-    ret = eval_script('django_import_finder.py', env=env)
+    python_path = os.getenv('PYTHONPATH', '').split(os.pathsep)
+    python_path.extend(pths)
+    python_path = os.pathsep.join(python_path)
+
+    django_settings_module = os.getenv('DJANGO_SETTINGS_MODULE', os.path.basename(django_root_dir) + '.settings')
+    if env is None:
+        env = {}
+    env.setdefault('DJANGO_SETTINGS_MODULE', django_settings_module)
+    env.setdefault('PYTHONPATH', python_path)
```

Comment:
Good point, checking first if it is necessary is definitely better.
As for env argument.. the idea of running scripts in a separate interpreter and read what they have printed to stdout suggests that we should be able to define the environment in which we want them to execute. So I thought an argument like "env" would come handy when calling eval_script_with_django.